src/services/TestResultsService.ts

```diff
@@ -364,7 +366,7 @@ export class TestResultsService {
   }
 
   private lastDayOfMonthInNextYear(inputDate: Date): Date {
-    return dateFns.lastDayOfMonth(dateFns.addYears(inputDate, 1));
+    return dateFns.endOfDay(dateFns.lastDayOfMonth(dateFns.addYears(inputDate, 1)));
```

Comment:
required as date-fns can't differentiate between UTC and local times, so when "lastDayOfTheMonth" sets the time to 00:00:000 BST (as it does), .toISOString winds that back an hour to the day before in UTC time. Circumventing this properly would require a dangerous upgrade from date-fns 1.X to 2.X which no longer accepts string inputs and everything gets really really complicated.